### PR TITLE
fix shortestPath validation

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: tests/flow/test_shortest_path.py [hunk 1]
```diff
@@ -135,3 +135,11 @@ def test06_restricted_reltypes(self):
         # The longer traversal will be found
         expected_result = [[1], [2], [3], [4]]
         self.env.assertEqual(actual_result.result_set, expected_result)
+
+    def test07_shortestPath_in_filter(self):
```
Comment (swilly22) [inline tests/flow/test_shortest_path.py:139]: I think we need to cover cases where the path in `shortestPath` contains both bounded and unbounded entities
e.g.
```
shortestPath((a)-[:E|:E2*]->()
shortestPath(()-[:E|:E2*]->(b)
shortestPath(()-[:E|:E2*]->()
```

In addition to new aliases
`shortestPath((z)-[:E|:E2*]->(x)`

Moreover: `shortestPath(({v:1})-[:E|:E2*]->()`

File: src/ast/ast_validations.c [hunk 2]
```diff
@@ -64,6 +64,43 @@ static bool _ValidateAllShortestPaths
...
+	if(t == CYPHER_AST_SHORTEST_PATH &&
+	   cypher_ast_shortest_path_is_single(root)) {
+		return true;
+	}
+
+	// shortestPaths is invalid in the MATCH pattern
+	if(t == CYPHER_AST_MATCH) {
+		const cypher_astnode_t *pattern = cypher_ast_match_get_pattern(root);
+		return pattern != NULL && _ValidateShortestPaths(pattern);
+	}
```
Comment (swilly22) [inline src/ast/ast_validations.c:85]: comment states that `shortestPaths` is INVALID in `MATCH` but we return true if there's a `pattern` and the `pattern` is `CYPHER_AST_SHORTEST_PATH`

File: src/ast/ast_validations.c [hunk 3]
```diff
@@ -64,6 +64,43 @@ static bool _ValidateAllShortestPaths
 	return false;
 }
 
+// validate that shortestPaths is in a supported place
+static bool _ValidateShortestPaths
```
Comment (swilly22) [inline src/ast/ast_validations.c:68]: Seems like this function would traverse the entire AST, is that correct ?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/ast/ast_validations.c (hunk 2 above)
Comment (AviAvni) [reply src/ast/ast_validations.c:85]: true is invalid in this function like the function above and it is invalid only on the pattern part not the predicate

File: src/ast/ast_validations.c (hunk 3 above)
Comment (AviAvni) [reply src/ast/ast_validations.c:68]: yes it is a copy of the function above _ValidateAllShortestPaths

File: tests/flow/test_shortest_path.py (hunk 1 above)
Comment (AviAvni) [reply tests/flow/test_shortest_path.py:139]: done

**Review by swilly22 — DISMISSED** [review]
looks good, one simplification

**Pull request merged**